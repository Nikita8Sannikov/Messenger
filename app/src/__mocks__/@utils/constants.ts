export const DEV = true;
export const BASE_URL = "https://mocked-url.com";
export const API_URL = "https://mocked-api-url.com";
export const SOCKET_URL = "wss://mocked-socket-url.com";
export const COOKIE_NAME = "mocked-cookie-name";
export const MAIL_FEEDBACK = "mocked@mail.com";
export const TIMEOUT_IS_WRITE_MESSAGE = 5000;
export const TIMEOUT_HIDE_UPPER_DATE = 3000;
export const AXIOS_RESPONSE_ENCODING = "utf-8";
export const AXIOS_TIMEOUT = 10000;
export const SOCKET_RECONECTION_ATTEMPTS = 3;
export const SOCKET_RECONNECTION_DELAY = 1000;
export const SOCKET_ACK_TIMEOUT = 500;
export const ARRAY_LOGS_LENGTH = 100;
export const LOGS_FILE_NAME = "logs.txt";
export const NO_PHOTO = "/assets/images/noPhoto.jpg";
export const REQUIRED_FIELD = "This field is required";
export const MY_ID = "me";
