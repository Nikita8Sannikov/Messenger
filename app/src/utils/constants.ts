import i18next from "@service/i18n";

export const CLIENT_URL = import.meta.env.VITE_CLIENT_URL as string;
export const SERVER_URL = import.meta.env.VITE_SERVER_URL as string;
export const COOKIE_NAME = import.meta.env.VITE_COOKIE_NAME as string;
export const SOCKET_IO_CLIENT = import.meta.env.VITE_SOCKET_IO_CLIENT as string;
export const MAIL_FEEDBACK = import.meta.env.VITE_MAIL_FEEDBACK as string;
export const TIMEOUT_IS_WRITE_MESSAGE = parseInt(import.meta.env.VITE_TIMEOUT_IS_WRITE_MESSAGE as string);
export const TIMEOUT_HIDE_UPPER_DATE = parseInt(import.meta.env.VITE_TIMEOUT_HIDE_UPPER_DATE as string);
export const AXIOS_RESPONSE_ENCODING = import.meta.env.REACT_APP_AXIOS_RESPONSE_ENCODING as string;
export const AXIOS_TIMEOUT = parseInt(import.meta.env.REACT_APP_AXIOS_TIMEOUT as string);
export const RECONECTION_ATTEMPTS = parseInt(import.meta.env.REACT_APP_RECONECTION_ATTEMPTS as string);
export const RECONNECTION_DELAY = parseInt(import.meta.env.REACT_APP_RECONNECTION_DELAY as string);

export const NO_PHOTO = "/assets/images/noPhoto.jpg";
export const REQUIRED_FIELD = i18next.t("utils.required_field");
export const MY_ID = "me";