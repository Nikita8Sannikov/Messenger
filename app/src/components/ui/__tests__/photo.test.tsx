import { fireEvent, render, screen } from "@testing-library/react";

import PhotoComponent from "@components/ui/photo";
import { NO_PHOTO } from "@utils/constants";

describe("PhotoComponent", () => {
	const mockProps = {
		src: "https://example.com/photo.jpg",
		alt: "Test Photo",
		showDeleteIcon: true,
		isLazy: false,
		clickHandler: jest.fn(),
		deleteHandler: jest.fn(),
	};

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("should render with image", () => {
		render(<PhotoComponent {...mockProps} />);
        
		const image = screen.getByTestId("image");
		expect(image).toBeInTheDocument();
		expect(image).toHaveAttribute("src", mockProps.src);
		expect(image).toHaveAttribute("alt", mockProps.alt);
	});

	it("should render with lazy loading", () => {
		render(
			<PhotoComponent 
				{...mockProps} 
				isLazy={true}
			/>,
		);
        
		const image = screen.getByTestId("image");
		expect(image).toHaveAttribute("loading", "lazy");
	});

	it("should render with eager loading by default", () => {
		render(<PhotoComponent {...mockProps} />);
        
		const image = screen.getByTestId("image");
		expect(image).toHaveAttribute("loading", "eager");
	});

	it("should not show delete icon on hover when showDeleteIcon is false", () => {
		render(
			<PhotoComponent 
				{...mockProps} 
				showDeleteIcon={false}
			/>,
		);
        
		const container = screen.getByTestId("image").parentElement;
		fireEvent.mouseEnter(container!);
        
		const closeIcon = screen.queryByTestId("close-icon");
		expect(closeIcon).not.toBeInTheDocument();
	});

	it("should not show delete icon for NO_PHOTO image", () => {
		render(
			<PhotoComponent 
				{...mockProps} 
				src={NO_PHOTO}
			/>,
		);
        
		const container = screen.getByTestId("image").parentElement;
		fireEvent.mouseEnter(container!);
        
		const closeIcon = screen.queryByTestId("close-icon");
		expect(closeIcon).not.toBeInTheDocument();
	});

	it("should call clickHandler when clicked", () => {
		render(<PhotoComponent {...mockProps} />);
        
		const container = screen.getByTestId("image").parentElement;
		fireEvent.click(container!);
        
		expect(mockProps.clickHandler).toHaveBeenCalled();
	});

	it("should hide delete icon when mouse leaves", () => {
		render(<PhotoComponent {...mockProps} />);
        
		const container = screen.getByTestId("image").parentElement;
		fireEvent.mouseEnter(container!);
		fireEvent.mouseLeave(container!);
        
		const closeIcon = screen.queryByTestId("close-icon");
		expect(closeIcon).not.toBeInTheDocument();
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<PhotoComponent {...mockProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
