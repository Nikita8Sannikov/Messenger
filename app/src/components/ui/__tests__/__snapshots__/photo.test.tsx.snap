// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="photo"
  >
    <img
      alt="Test Photo"
      class="photo__image"
      data-testid="image"
      loading="eager"
      src="https://example.com/photo.jpg"
    />
  </div>
</DocumentFragment>
`;
